storage/innobase/xtrabackup/test/suites/binlog/ib_slave_info.sh

```diff
@@ -74,7 +75,7 @@ run_cmd_expect_failure $XB_BIN $XB_ARGS --backup --slave-info --no-lock \
   --target-dir=$topdir/backup
 
 vlog "Full backup of the slave server"
-xtrabackup --backup --lock-ddl=false --target-dir=$topdir/backup --slave-info 2>&1 | tee $topdir/pxb.log
+xtrabackup --backup --lock-ddl=false --target-dir=$topdir/backup --slave-info --safe-slave-backup 2>&1 | tee $topdir/pxb.log
```

Comment:
Is it possible for xtrabackup to query the server and do the right thing?

Reply:
possible yes, but --safe-slave-backup can be considerer intrusive as it will stop replication applier thread. IMHO the user should be the one adding this.